source/deploy/standalone/dev_ssl.md.erb

```diff
@@ -0,0 +1,845 @@
+        </td>
+    </tr>
+    <tr>
+        <td>
+            safari
+        </td>
+        <td>
+            9.1.1
+        </td>
+        <td>
+            <pre class="highlight">N/A</pre>
+        </td>
+        <td>
+            <pre class="highlight"><span class="prompt">$ </span>system_profiler -xml SPApplicationsDataType | xmllint --xpath \
+"//array/dict/array/dict[string='Safari']/key[.='version']/following-sibling::&#42;[1]/text()" -</pre>
+        </td>
+    </tr>
+</table>
+
+# Openssl configuration
+
+When setting up a dev environment with ssl it is better practice to use a self signed certificate, and leave your real certificate/key pair somewhere safe and encrypted, preferably on offline storage like a thumb drive in a safe place. In order to create a certificate that works and makes development convenient, you need to edit the `openssl.cnf` file.
+
+On OS X this is located at `/usr/local/etc/openssl/openssl.cnf` or if it isn't there you can copy `/usr/local/etc/openssl/openssl.cnf.default` to that path as a starting point, and edit from there.
+
+On linux it may be in one of the following locations: `/usr/local/ssl/openssl.cnf` `/usr/lib/ssl/openssl.cnf` `/etc/ssl/openssl.cnf` `/etc/pki/tls/openssl.cnf`
+
+the changes you need to make are as follows:
+
+## uncomment these lines, they enable needed functionality:
+~~~
+# unique_subject = no
+# copy_extensions = copy
+# req_extensions = v3_req
+# keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+# keyUsage = cRLSign, keyCertSign
+~~~
+
+## comment out these lines, they get in the way:
+~~~
+attributes              = req_attributes
+[ req_attributes ]
+challengePassword               = A challenge password
+challengePassword_min           = 4
+challengePassword_max           = 20
+unstructuredName                = An optional company name
+~~~
+
+## and modify the following values as indicated:
+~~~
+default_days = 365 → 3650 # there's no reason to have to redo this every year
+default_md = default → sha256 # default in openssl 1.1.0, needed for modern browsers
+~~~
+
+## modify the following values as desired:
+~~~
+countryName_default             = # two letter country code
+stateOrProvinceName_default     = # name of province or state
+localityName_default            = # name of city
+0.organizationName_default      = # name of organization
+organizationalUnitName_default  = # name of department
+commonName_default              = # your main domain
+emailAddress_default            = # your email
+~~~
+
+## add this on a new line under `keyUsage = nonRepudiation, digitalSignature, keyEncipherment`:
+~~~
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = domain1.dev
+DNS.2 = localhost
+DNS.3 = domain2.dev
+DNS.4 = &#42;.domain1.dev
+# and as many more as you want, note: you must use domains not IP addresses
+~~~
+
+## A working sample is provided below, replace occurences of phusion with your own domain:
```

Comment:
"occurrences"
better to pre-replace phusion with something like "exampledomain1", and consistently use it throughout the previous examples too (exampledomain2 etc. for the alt names).